dsdl_compiler/libcanard_dsdl_compiler/data_type_template.tmpl

```diff
@@ -0,0 +1,128 @@
+/*
+ * UAVCAN data structure definition for libcanard.
+ *
+ * Autogenerated, do not edit.
+ *
+ * Source file: ${t.source_file}
+ */
+
+#ifndef ${t.include_guard}
+#define ${t.include_guard}
+
+#include <stdint.h>
+#include "canard.h"
+
+#ifndef FLOAT16
+typedef uint16_t float16;
+#else
+typedef FLOAT16 float16;
+#endif
+
+#ifndef SATURATE
+#define SATURATE(x, max) ( ((x) > max) ? max : ( (-(x) > max) ? (-max) : (x) ) );
```

Comment:
I think if you want to expose macros in HEADERS, you should prefix them with CANARD as done in canard.h